### PR TITLE
repl: split auxiliary output from kore output

### PR DESCRIPTION
**Pull request by ana-pantilie**

### CODE REVIEW

File: kore/src/Kore/Repl.hs [hunk 1]
```diff
@@ -103,7 +103,10 @@ runRepl axioms' claims' logger replScript replMode outputFile = do
         void
             $ flip evalStateT st
             $ flip runReaderT config
-            $ replInterpreter printIfNotEmpty cmd
+            $ replInterpreter
```
Comment (vladciobanu) [inline kore/src/Kore/Repl.hs]: I think the internals are leaking out a bit here with these changes. How about reverting this change, and renaming `replInterpreter` to `replInterpreter0` or `replInterpreter'` (and renaming all internal occurrences to that), then writing `replInterpreter fn cmd = replInterpreter0 (PrintAuxOutput fn) (PrintKoreOutput fn) cmd`?

File: kore/src/Kore/Repl/Data.hs [hunk 2]
```diff
@@ -507,3 +500,39 @@ data GraphProofStatus
...
+    :: forall m a
+    .  MonadSimplify m
+    => UnifierWithExplanation m a
+    -> m (Either ReplOutput (NonEmpty a))
+runUnifierWithExplanation (UnifierWithExplanation unifier) =
+    either explainError failWithExplanation <$> unificationResults
+  where
+    unificationResults
+        ::  m
+                (Either UnificationOrSubstitutionError ([a], First ReplOutput))
```
Comment (vladciobanu) [inline kore/src/Kore/Repl/Data.hs]: I know you didn't make this change, but this line looks weird. I think you should move it up in-line with the `m`.

File: kore/src/Kore/Repl/Interpreter.hs [hunk 3]
```diff
@@ -652,6 +665,17 @@ labelDel lbl = do
        else
            putStrLn' "Label doesn't exist."
 
+outputsKore :: ReplCommand -> Bool
```
Comment (vladciobanu) [inline kore/src/Kore/Repl/Interpreter.hs]: I'm not sure we need this. I would say that commands that don't output patterns should not go through pipes. We can go back on this later.

File: kore/src/Kore/Repl/Interpreter.hs [hunk 4]
```diff
@@ -893,26 +933,37 @@ appendTo
...
+
+appendCommand
+    :: forall claim m
+    .  Claim claim
+    => MonadSimplify m
+    => MonadIO m
+    => ReplCommand
+    -> FilePath
+    -> ReplM claim m ()
+appendCommand cmd file = do
```
Comment (vladciobanu) [inline kore/src/Kore/Repl/Interpreter.hs]: I think redirect/append should write everything, not just patterns.

File: kore/src/Kore/Repl/Data.hs [hunk 5]
```diff
@@ -119,6 +122,20 @@ newtype RuleName = RuleName
     { unRuleName :: String
     } deriving (Eq, Show)
 
+newtype ReplOutput =
+    ReplOutput
+    { unReplOutput :: [ReplOut]
+    } deriving (Eq, Show, Semigroup, Monoid)
+
+newtype PrintAuxOutput = PrintAuxOutput
```
Comment (ttuegel) [inline kore/src/Kore/Repl/Data.hs]: Could we add comments here explaining how `PrintAuxOutput` and `PrintKoreOutput` are generally used? It wasn't clear to me, and I had to scroll most of the way through the pull request to discover that `KoreOut` is piped through `kast`.

**Pull request merged**